BaragonUI/app/components/groupDetail/GroupDetail.jsx

```diff
@@ -1,36 +1,116 @@
+      <div>
+        <div className="row">
+          <GroupTitleBar
+            group={this.props.group}
+            domain={this.props.domain}
+            targetCount={this.props.targetCount}
+            editable={this.props.editable}
+            afterModifyTargetCount={this.props.afterModifyTargetCount}
+          />
+         </div>
+         <div className="row">
+           <GroupDomains
+             domains={this.props.domainsServed}
+             defaultDomain={this.props.domain}
+           />
+        </div>
+        <div className="row">
+          <GroupTrafficSources
+            trafficSources={this.props.trafficSources}
+            group={this.props.group}
+            editable={this.props.editable}
+            afterAddTrafficSource={this.props.afterAddTrafficSource}
+            afterRemoveTrafficSource={this.props.afterRemoveTrafficSource}
+          />
+        </div>
+        <div className="row">
+          <GroupAgents
+            agents={this.props.agents}
+          />
+          <GroupKnownAgents
+            knownAgents={this.props.knownAgents}
+            group={this.props.group}
+            editable={this.props.editable}
+            afterRemoveKnownAgent={this.props.afterRemoveKnownAgent}
+          />
+        </div>
+        <div className="row">
+          <GroupBasePaths
+            basePaths={this.props.basePaths}
+            domain={this.props.domain}
+            group={this.props.group}
+            editable={this.props.editable}
+            afterRemoveBasePath={this.props.afterRemoveBasePath}
+          />
+        </div>
+      </div>
+    );
+  }
 }
 
-export default connect((state, ownProps) => ({
-  group: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data']),
+const mapStateToProps = (state, ownProps) => ({
+  domain: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data', 'domain']),
+  domainsServed: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data', 'domains']),
+  group: ownProps.params.groupId,
   basePaths: Utils.maybe(state, ['api', 'basePaths', ownProps.params.groupId, 'data']),
   targetCount: Utils.maybe(state, ['api', 'targetCount', ownProps.params.groupId, 'data']),
   agents: Utils.maybe(state, ['api', 'agents', ownProps.params.groupId, 'data']),
   knownAgents: Utils.maybe(state, ['api', 'knownAgents', ownProps.params.groupId, 'data']),
-  trafficSources: Utils.maybe(state, ['api', 'trafficSources', ownProps.params.groupId, 'data'])
-}))(rootComponent(GroupDetail, (props) => refresh(props.params.groupId)));
\ No newline at end of file
+  trafficSources: Utils.maybe(state, ['api', 'group', ownProps.params.groupId, 'data', 'trafficSources']),
+  editable: window.config.allowEdit,
+});
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  afterModifyTargetCount: () => dispatch(FetchGroupTargetCount.trigger(ownProps.params.groupId)),
+  afterAddTrafficSource: () => dispatch(FetchGroup.trigger(ownProps.params.groupId)),
```

Comment:
it'd probably be clearer to name these by what function they are performing, not by the context they are being used in.

i.e. `fetchGroup` which can then be used in both the `then` for adding and removing traffic source, rather than defining those separately.

You can also do something a bit more efficient for ones with multiple calls (like the `afterRemoveKnownAgent` currently is). Dispatch returns a promise. So, rather that triggering one, waiting for response, and triggering another, we could do something like
```
fetchAllAgents: () => Promise.all([
  dispatch(FetchGroupKnownAgents.trigger(ownProps.params.groupId),
  dispatch(FetchGroupAgents.trigger(ownProps.params.groupId)
])
```
The Promise.all returns a single promise that is completed when all of the contained promises return